src/Action/CancelAction.php

```diff
@@ -55,9 +56,9 @@ public function supports($request): bool
     /**
      * @throws PayUException
      */
-    private function canCancelPayment(Model $model, PaymentInterface $payment): bool
+    private function canCancelPayment(string $orderId, ?string $configKey): bool
     {
-        $response = $this->orderRequestService->retrieve($model->orderId(), PaymentHelper::getConfigKey($model, $payment));
```

Comment:
Here the `orderId` from the model was used, but it does not always have to exist, `PaymentHelper::getOrderId` tries to take the `orderId` from `firstModel` in the first place